### PR TITLE
Rpc enhancements

### PR DESCRIPTION
**Pull request by boqiu**
1. Add rpc `admin_terminateSync` to terminate file/chunks sync.
2. When p2p sync failed, do not allow to resync if goal changed.
3. Support to configure separate endpoint for admin and mine rpc.

- - -
This change is [Reviewable](https://reviewable.io/reviews/zero-gravity-labs/zerog-storage-rust/15)

### CODE REVIEW

File: node/sync/src/service.rs [hunk 1]
```diff
@@ -576,6 +576,21 @@ impl SyncService {
...
+                        bail!("Cannot change failed file sync to chunks sync");
+                    }
+                }
+                None => {
+                    if !goal.is_all_chunks() {
+                        bail!("Cannot change failed chunks sync to file sync");
+                    }
+                }
+            }
+
```
Comment (MiniFrenchBread) [inline node/sync/src/service.rs:593]: Update sync goal here?

File: node/sync/src/service.rs [hunk 2]
```diff
@@ -576,6 +576,21 @@ impl SyncService {
...
             controller.reset();
         }
```
Comment (MiniFrenchBread) [inline node/sync/src/service.rs:596]: Update controller's range if the old status is completed, the old goal is not all chunks and the new goal is different from the old.

File: node/sync/src/service.rs (hunk 1 above)
Comment (MiniFrenchBread) [inline node/sync/src/service.rs:593]: Should we update controller's range if old status is failed?

**Review by boqiu — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/zero-gravity-labs/zerog-storage-rust/15)* status: 0 of 11 files reviewed, 2 unresolved discussions (waiting on @MiniFrenchBread and @peilun-conflux)

___
*[`node/sync/src/service.rs` line 593 at r1](https://reviewable.io/reviews/zero-gravity-labs/zerog-storage-rust/15#gh-1469677434:-NpOEpmA5BsUDq6Zqou4:b-896fix) ([raw file](https://github.com/zero-gravity-labs/zerog-storage-rust/blob/6ff03e98d7afbba4c5ad7acefedca0e7b87ced67/node/sync/src/service.rs#L593)):*
<details><summary><i>Previously, MiniFrenchBread wrote…</i></summary><blockquote>

Should we update controller's range if old status is failed? 
</blockquote></details>

Done.
___
*[`node/sync/src/service.rs` line 596 at r1](https://reviewable.io/reviews/zero-gravity-labs/zerog-storage-rust/15#gh-1469672704:-NpOEwB7EJwM7ADtFBQU:b-896fix) ([raw file](https://github.com/zero-gravity-labs/zerog-storage-rust/blob/6ff03e98d7afbba4c5ad7acefedca0e7b87ced67/node/sync/src/service.rs#L596)):*
<details><summary><i>Previously, MiniFrenchBread wrote…</i></summary><blockquote>

Update controller's range if the old status is completed, the old goal is not all chunks and the new goal is different from the old.
</blockquote></details>

Done.

**Review by MiniFrenchBread — APPROVED** [review]

**Pull request merged**